scripts/check-ancient-pip.sh

```diff
@@ -7,19 +7,19 @@ PYTHON=$1
 
 BROKEN_VIRTUALENV=$($PYTHON -c'import tempfile; print(tempfile.mkdtemp())')
 
-trap "rm -rf $BROKEN_VIRTUALENV" EXIT
+trap 'rm -rf $BROKEN_VIRTUALENV' EXIT
```

Comment:
1. This is the fix recommended by spellcheck, so I think it’s legit.

2. The variable remains in scope, as this example demonstrates:

    ```console
    $ cat traptest.sh
    #!/usr/bin/env bash
    MESSAGE="hello world"
    trap 'echo $MESSAGE' EXIT
    sleep 2
    exit 1

    $ ./traptest.sh
    hello world
    ```

3. It uses the definition at the point of trap execution, not definition. So you can redefine variables used in the trap:

    ```console
    $ cat traptest.sh
    #!/usr/bin/env bash
    MESSAGE="hello world"
    trap 'echo $MESSAGE' EXIT
    MESSAGE="foo bar"
    sleep 2
    exit 1

    $ ./traptest.sh
    foo bar
    ```

    That doesn’t affect either of our two uses and in practice I think is fine because most of this code is not long for this world, and I can’t be bothered to work out how to stop shellcheck nagging about these lines.